### PR TITLE
wifi-subsys: add main of wifi subsys

### PR DESCRIPTION
**Pull request by xkevin190**
### Contribution description

in this PR was added the implementation and initialization of all components created for wifi subsys.

this pr still in progress only missing add sntp settings i have some problems with the internet and i can not continue. please see the code and share with me your opinions.

### CODE REVIEW

**Comment by luisan00** [comment]
some warnings of static-test need to be resolved before to review

**Comment by luisan00** [comment]
squash please

**Comment by luisan00** [comment]
some conflicts need to be solved :+1:

**Review by luisan00 — COMMENTED** [review]
please squash before to continue

**Comment by xkevin190** [comment]
@luisan00 i think will do the other stuff that we need for finish main, when the boys finished with his small fixes i added his PRs to my branch

**Comment by xkevin190** [comment]
i will work over this branch.

**Comment by luisan00** [comment]
> i will work over this branch.

This branch is being inconsistent in relation to base branch, so first of all i recommend merge it

File: wifi-subsys/main/default_params.h [hunk 1]
```diff
@@ -59,8 +59,8 @@ extern "C"
  * @{
  */
 #define WSTA                            /*!< NVS namespace for WiFi station mode settings */
-#define WSTA_SSID "Chicho"              /*!< Default SSID for the station mode */
-#define WSTA_PASS "Pirulin0312"           /*!< Default password for station */
+#define WSTA_SSID "IPhone de Kevin"              /*!< Default SSID for the station mode */
+#define WSTA_PASS "kevin1994"           /*!< Default password for station */
```
Comment (eduazocar) [inline wifi-subsys/main/default_params.h:63]: Fix all the default params to m4a-sta (SSID) and change-me (PASSWORD)

File: wifi-subsys/components/network/src/wifi.c [hunk 2]
```diff
@@ -749,12 +749,6 @@ esp_err_t wifi_init(void)
...
 
     if (!first_boot) {
-
-        if (nvs_init() != ESP_OK)
-        {
-            ESP_LOGE(__func__, "Starting the [N]on [V]olarile [S]torage");
-        }
-
         err = esp_netif_init();
         if(err != ESP_OK){
```
Comment (eduazocar) [inline wifi-subsys/components/network/src/wifi.c:753]: if required a simple space before ()

Comment (eduazocar) [reply wifi-subsys/components/network/src/wifi.c:753]: There are many if's with the same format error presented. please, fix them

**Review by eduazocar — CHANGES_REQUESTED** [review]
Well, You did a great job, but this need some changes previously to merge it

File: wifi-subsys/main/default_params.h [hunk 3]
```diff
@@ -70,10 +70,20 @@ extern "C"
...
+#define WIFI_ENABLED 1
 #define WIFI_MODE
-#define DEFAULT_SERVER_TO_PING  "www.google.com"
 /** @} */
 
+/**
+ * @name internet connection settings
+ * @{
+ */
+#define I_CONECTION
```
Comment (luisan00) [inline wifi-subsys/main/default_params.h:81]: do you mean:
```suggestion
#define I_CONNECTION
```
?

**Pull request merged**